KarhooUISDK/Controllers/Trips/KarhooTripsProvider.swift

```diff
@@ -46,34 +46,38 @@ public final class KarhooTripsProvider: TripsProvider {
     }
     
     public func stop() {
-        self.reachability.remove(listener: self)
-        self.timer.invalidate()
+        reachability.remove(listener: self)
+        itemsOffset = 0
+        timer.invalidate()
     }
 
     deinit {
         stop()
     }
 
     fileprivate func loadTrips() {
-        let tripSearch = TripSearch(tripStates: TripStatesGetter().getStatesForTripRequest(type: requestType),
-                                    paginationRowCount: pageSize,
-                                    paginationOffset: itemsOffset)
-
-        tripService.search(tripSearch: tripSearch)
-                    .execute(callback: { [weak self] (result: Result<[TripInfo]>) in
-            guard result.isSuccess() else {
-                self?.delegate?.tripProviderFailed(error: result.errorValue())
-                return
-            }
-
-            guard let trips = result.successValue() else {
-                self?.delegate?.fetched(trips: [])
-                return
-            }
+        let tripSearch = TripSearch(
+            tripStates: TripStatesGetter().getStatesForTripRequest(type: requestType),
+            paginationRowCount: pageSize,
+            paginationOffset: itemsOffset
+        )
         
-            self?.noMoreItems = trips.count == 0 ? true : false
-            self?.delegate?.fetched(trips: trips)
-        })
+        tripService
+            .search(tripSearch: tripSearch)
+            .execute(callback: { [weak self] (result: Result<[TripInfo]>) in
+                guard result.isSuccess() else {
+                    self?.delegate?.tripProviderFailed(error: result.errorValue())
+                    return
+                }
+                
+                guard let trips = result.successValue() else {
+                    self?.delegate?.fetched(trips: [])
+                    return
+                }
+                
+                self?.noMoreItems = trips.count == 0 ? true : false
```

Comment:
`? true : false` is redundant. `self?.noMoreItems = trips.count == 0` is enough to achieve the desired behavior.

Reply:
I want to suggest even more simple, easy to read and developer-friendly solution: 
`self?.noMoreItems = trips.isEmpty`